pkg/daemon/ceph/osd/encryption.go

```diff
@@ -160,6 +162,17 @@ func dumpLUKS(context *clusterd.Context, disk string) (string, error) {
 	return cryptsetupOut, nil
 }
 
+func openLUKS(context *clusterd.Context, disk, target, passphrase string) error {
+	args := []string{"luksOpen", "--verbose", "--allow-discards", disk, target}
+	err := context.Executor.ExecuteCommandWithStdin(luksOpenCmdTimeOut, cryptsetupBinary, &passphrase, args...)
+	if err != nil {
+		logger.Errorf("failed to open LUKS device %q", disk)
```

Comment:
Why both log and return err? The error is going to be surfaced eventually so just returning a wrapped error seems sufficient.